Services/UserService.cs

```diff
@@ -5,23 +5,23 @@
 using System;
 using Microsoft.Extensions.Options;
 using Microsoft.Extensions.Logging;
-using Aserto.AspNetCore.Middleware.Clients;
+using Aserto.AspNetCore.Middleware.Clients.Directory.V3;
 using Aserto.AspNetCore.Middleware.Options;
 using Aserto.TodoApp.Options;
 
 namespace Aserto.TodoApp.Services
 {
     public class UserService : IUserService
     {
-        private readonly DirectoryAPIClient directoryClient;
+        private readonly Aserto.AspNetCore.Middleware.Clients.Directory.V3.Directory directoryClient;
         private readonly DirectoryConfig opts;
 
         public UserService(IOptions<DirectoryConfig> config)
         {
             this.opts = config.Value;
             if (!opts.IsValid())
             {
-                throw new Exception("Invalid config");
+                throw new Exception("Invalid config. Service url can not be empty");
```

Comment:
How do we know that the service URL is empty? Couldn't there be some other reason why `IsValid()` returns false?

Reply:
Is the only thing that we check